library/std/src/sys_common/once/queue.rs

```diff
@@ -121,6 +122,16 @@ impl Once {
         self.state_and_queue.load(Ordering::Acquire).addr() == COMPLETE
     }
 
+    #[inline]
+    pub(crate) fn state(&mut self) -> ExclusiveState {
+        match self.state_and_queue.get_mut().addr() {
+            INCOMPLETE => ExclusiveState::Incomplete,
+            POISONED => ExclusiveState::Poisoned,
+            COMPLETE => ExclusiveState::Complete,
```

Comment:
Not necessary for this PR, but I wonder if we can convert `INCOMPLETE`, `POISONED`, and `COMPLETE` to an enum.

Reply:
Since there is no generic `Atomic<T>` type (so the `enum` would not be used directly) and `as` casts may not be used in `match` guards, the current way has less boilerplate.